portal-api/api-tests/v1/utilisation-report-service/fee-record-corrections/get-completed-fee-record-corrections.api-test.js

```diff
@@ -0,0 +1,98 @@
+const { as, get } = require('../../../api.js')(app);
+const testUserCache = require('../../../api-test-users.js');
+const { withClientAuthenticationTests } = require('../../../common-tests/client-authentication-tests.js');
+const { withRoleAuthorisationTests } = require('../../../common-tests/role-authorisation-tests.js');
+const { PAYMENT_REPORT_OFFICER } = require('../../../../src/v1/roles/roles.js');
+
+console.error = jest.fn();
+
+describe('GET /v1/banks/:bankId/utilisation-reports/completed-corrections', () => {
+  const getUrl = ({ bankId }) => `/v1/banks/${bankId}/utilisation-reports/completed-corrections`;
+
+  let testUsers;
+  let barclaysBank;
+  let aBarclaysPaymentReportOfficer;
+
+  const exporter = 'An exporter';
+  const correctionId = 7;
+  const dateCorrectionReceived = new Date('2024-01-01');
+  const oldFacilityId = '11111111';
+  const newFacilityId = '22222222';
+  const bankCommentary = 'Some bank commentary';
+
+  beforeAll(async () => {
+    await SqlDbHelper.initialize();
+    await SqlDbHelper.deleteAllEntries('UtilisationReport');
+
+    testUsers = await testUserCache.initialise(app);
+    aBarclaysPaymentReportOfficer = testUsers().withRole(PAYMENT_REPORT_OFFICER).withBankName('Barclays Bank').one();
+    barclaysBank = aBarclaysPaymentReportOfficer.bank;
+
+    const aUtilisationReport = new UtilisationReportEntityMockBuilder().withBankId(barclaysBank.id).build();
+
+    const aFeeRecord = FeeRecordEntityMockBuilder.forReport(aUtilisationReport)
+      .withExporter(exporter)
+      .withFacilityId(oldFacilityId)
+      .withStatus(FEE_RECORD_STATUS.TO_DO_AMENDED)
+      .build();
+    aUtilisationReport.feeRecords = [aFeeRecord];
+
+    const aFeeRecordCorrection = FeeRecordCorrectionEntityMockBuilder.forFeeRecordAndIsCompleted(aFeeRecord, true)
+      .withId(correctionId)
+      .withDateReceived(dateCorrectionReceived)
+      .withReasons([RECORD_CORRECTION_REASON.FACILITY_ID_INCORRECT, RECORD_CORRECTION_REASON.OTHER])
+      .withPreviousValues({
+        facilityId: oldFacilityId,
+      })
+      .withCorrectedValues({
+        facilityId: newFacilityId,
+      })
+      .withBankCommentary(bankCommentary)
+      .build();
+
+    await SqlDbHelper.saveNewEntries('UtilisationReport', [aUtilisationReport]);
+    await SqlDbHelper.saveNewEntries('FeeRecordCorrection', [aFeeRecordCorrection]);
+  });
+
+  withClientAuthenticationTests({
+    makeRequestWithoutAuthHeader: () => get(getUrl({ bankId: barclaysBank.id })),
+    makeRequestWithAuthHeader: (authHeader) => get(getUrl({ bankId: barclaysBank.id }), { headers: { Authorization: authHeader } }),
+  });
+
+  withRoleAuthorisationTests({
+    allowedRoles: [PAYMENT_REPORT_OFFICER],
+    getUserWithRole: (role) => testUsers().withRole(role).withBankName(barclaysBank.name).one(),
+    makeRequestAsUser: (user) => as(user).get(getUrl({ bankId: barclaysBank.id })),
+    successStatusCode: HttpStatusCode.Ok,
+  });
+
```

Comment:
Is this where we would typically put a test to check that we return unauthorised if the users bank and the bank id in the params do not match?